File: public/scripts/patch-restricoes.py
import re

with open('foco-02.js', 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Update HTML structure for Restrições
target_html1 = '''          <!-- ==================== RESTRIÇÕES ==================== -->
          <h4 style="margin: 24px 0 16px 0; color: #0056b3; border-bottom: 2px solid #ddd; padding-bottom: 8px;">
            Restrições
          </h4>

          <div class="form-group editavel">
            <label>Restrições verificadas:</label>

            <div class="checkbox-group" id="group-restricoes_${rip}" class="checkbox-group">'''

replacement_html1 = '''          <!-- ==================== RESTRIÇÕES ==================== -->
          <h4 style="margin: 24px 0 16px 0; color: #0056b3; border-bottom: 2px solid #ddd; padding-bottom: 8px;">
            Restrições
          </h4>

          <div class="form-group editavel">
            <label>Há restrições e condições limitadoras?</label>
            <div class="radio-group" style="display: flex; gap: 15px; align-items: center; margin-top: 5px;">
              <label class="radio-option">
                <input type="radio" name="imoveis[${index}][tem_restricoes]" value="Sim" required />
                Sim
              </label>
              <label class="radio-option">
                <input type="radio" name="imoveis[${index}][tem_restricoes]" value="Não" />
                Não
              </label>
              <label class="radio-option">
                <input type="radio" name="imoveis[${index}][tem_restricoes]" value="Não há informação suficiente" />
                Não há informação suficiente
              </label>
            </div>

            <div id="bloco-restricoes-detalhes_${rip}" style="display: none; flex-direction: column; gap: 8px; margin-top: 15px; padding-left: 15px; border-left: 3px solid #ccc;">
              <label>Restrições verificadas:</label>

              <div class="checkbox-group" id="group-restricoes_${rip}" class="checkbox-group">'''

target_html2 = '''            <div id="bloco-obs-restricoes_${rip}" style="display: none; flex-direction: column; gap: 6px; margin-top: 16px;">
              <label for="obs-restricoes_${rip}">Observações sobre as restrições:</label>
              <textarea
                id="obs-restricoes_${rip}"
                name="imoveis[${index}][obs_restricoes]"
                placeholder="Descreva informações complementares sobre as restrições verificadas..."
              ></textarea>
            </div>
          </div>'''

replacement_html2 = '''            <div id="bloco-obs-restricoes_${rip}" style="display: none; flex-direction: column; gap: 6px; margin-top: 16px;">
              <label for="obs-restricoes_${rip}">Observações sobre as restrições:</label>
              <textarea
                id="obs-restricoes_${rip}"
                name="imoveis[${index}][obs_restricoes]"
                placeholder="Descreva informações complementares sobre as restrições verificadas..."
              ></textarea>
            </div>
            </div> <!-- Fecha bloco-restricoes-detalhes -->
          </div>'''

if target_html1 in content:
    content = content.replace(target_html1, replacement_html1)
    print("HTML1 restricoes applied.")
if target_html2 in content:
    content = content.replace(target_html2, replacement_html2)
    print("HTML2 restricoes applied.")

# Regex to remove "Nenhuma restrição identificada" if it exists
remove_pattern = r'<label class="checkbox-option">\s*<input[^>]*value="Nenhuma restrição identificada"[^>]*>\s*Nenhuma restrição identificada\s*</label>'
content = re.sub(remove_pattern, '', content)

# 2. Add JavaScript logic
target_js = "populateChecks('restricoes', dados.restricoes);"

replacement_js = '''populateChecks('restricoes', dados.restricoes);

        // Lógica condicional para Restrições
        const restricaoRadios = bloco.querySelectorAll(`input[name="imoveis[${index}][tem_restricoes]"]`);
        const blocoRestricoesDetalhes = document.getElementById(`bloco-restricoes-detalhes_${rip}`);
        const blocoObsRestricoes = document.getElementById(`bloco-obs-restricoes_${rip}`);
        const restricoesCheckboxes = bloco.querySelectorAll(`input[name="imoveis[${index}][restricoes][]"]`);

        function updateTemRestricoes() {
            let val = '';
            restricaoRadios.forEach(r => { if (r.checked) val = r.value; });
            if (blocoRestricoesDetalhes) {
                blocoRestricoesDetalhes.style.display = (val === 'Sim') ? 'flex' : 'none';
            }
        }
        restricaoRadios.forEach(r => r.addEventListener('change', updateTemRestricoes));

        function updateObsRestricoes() {
            let hasChecked = false;
            restricoesCheckboxes.forEach(c => {
                if (c.checked && !c.value.toLowerCase().includes('nenhuma')) hasChecked = true;
            });
            if (blocoObsRestricoes) {
                blocoObsRestricoes.style.display = hasChecked ? 'flex' : 'none';
            }
        }
        restricoesCheckboxes.forEach(c => c.addEventListener('change', updateObsRestricoes));

        // Inicialização com dados do banco
        if (dados.tem_restricoes) {
            restricaoRadios.forEach(r => { if (r.value === dados.tem_restricoes) r.checked = true; });
        } else if (dados.restricoes && Array.isArray(dados.restricoes) && dados.restricoes.length > 0) {
            restricaoRadios.forEach(r => { if (r.value === 'Sim') r.checked = true; });
        }
        updateTemRestricoes();
        updateObsRestricoes();'''

if target_js in content:
    content = content.replace(target_js, replacement_js)
    print("JS restricoes applied.")

with open('foco-02.js', 'w', encoding='utf-8') as f:
    f.write(content)

File: public/js/patch-restricoes.js
const fs = require('fs');

// ==========================================
// 1. PATCH HTML
// ==========================================
let html = fs.readFileSync('foco-02.html', 'utf8');

const htmlTarget = /<div class="form-group editavel">\s*<label>Restrições verificadas:<\/label>\s*<div class="checkbox-group">[\s\S]*?<\/div>\s*<\/div>/;

const htmlReplacement = `<div class="form-group editavel">
          <label>Restrições verificadas:</label>
          <div class="checkbox-group">
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Nenhuma restrição identificada" onchange="toggleRestricoes()" /> Nenhuma restrição identificada</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Faixa de fronteira" onchange="toggleRestricoes()" /> Faixa de fronteira</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Faixa de segurança" onchange="toggleRestricoes()" /> Faixa de segurança</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Faixa de domínio Ferrovia/Rodovia" onchange="toggleRestricoes()" /> Faixa de domínio Ferrovia/Rodovia</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Faixa de 100 metros ao longo da costa marítima" onchange="toggleRestricoes()" /> Faixa de 100 metros ao longo da costa marítima</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Circunferência de 1.320 metros em torno de instalações militares" onchange="toggleRestricoes()" /> Circunferência de 1.320 metros em torno de instalações militares</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Terra indígena" onchange="toggleRestricoes()" /> Terra indígena</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Território quilombola ou área de comunidade tradicional" onchange="toggleRestricoes()" /> Território quilombola ou área de comunidade tradicional</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Zona/Área de Interesse Social — ZEIS" onchange="toggleRestricoes()" /> Zona/Área de Interesse Social — ZEIS</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Área de segurança" onchange="toggleRestricoes()" /> Área de segurança</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Área Non Aedificandi" onchange="toggleRestricoes()" /> Área Non Aedificandi</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Restrição de uso/ocupação incidente sobre o imóvel" onchange="toggleRestricoes()" /> Restrição de uso/ocupação incidente sobre o imóvel</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Tombado como patrimônio histórico, artístico e/ou cultural" onchange="toggleRestricoes()" /> Tombado como patrimônio histórico, artístico e/ou cultural</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Poligonal de Porto Organizado" onchange="toggleRestricoes()" /> Poligonal de Porto Organizado</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Área operacional da RFFSA" onchange="toggleRestricoes()" /> Área operacional da RFFSA</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Ilha oceânica ou costeira sem sede de município" onchange="toggleRestricoes()" /> Ilha oceânica ou costeira sem sede de município</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Ilha fluvial ou lacustre" onchange="toggleRestricoes()" /> Ilha fluvial ou lacustre</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Localizada em loteamento" onchange="toggleRestricoes()" /> Localizada em loteamento</label>
              <label class="checkbox-option"><input type="checkbox" name="restricoes[]" value="Outra restrição identificada" onchange="toggleRestricoes()" /> Outra restrição identificada</label>
          </div>
          <div id="bloco-obs-restricoes" style="display: none; flex-direction: column; gap: 6px; margin-top: 8px;">
              <label for="obs_restricoes">Observações sobre as restrições:</label>
              <textarea id="obs_restricoes" name="obs_restricoes" placeholder="Descreva informações complementares sobre as restrições verificadas..."></textarea>
          </div>
      </div>`;

html = html.replace(htmlTarget, htmlReplacement);
fs.writeFileSync('foco-02.html', html);


// ==========================================
// 2. PATCH JS
// ==========================================
let js = fs.readFileSync('foco-02.js', 'utf8');

if (!js.includes('window.toggleRestricoes')) {
    const toggleFunc = `
window.toggleRestricoes = function() {
    const radios = document.querySelectorAll('input[name="restricoes[]"]');
    const hasChecked = Array.from(radios).some(r => r.checked);
    const blocoObs = document.getElementById('bloco-obs-restricoes');
    if (blocoObs) {
        blocoObs.style.display = hasChecked ? 'flex' : 'none';
    }
};
document.addEventListener('DOMContentLoaded', function () {`;
    js = js.replace("document.addEventListener('DOMContentLoaded', function () {", toggleFunc);
}

const target = "// Limpa o interval pois os dados já foram carregados";
const restricoesLogic = `
            // 7. RESTRIÇÕES
            const mockRestricoes = getArrayFieldFromLowestRip('restricoes');
            const mockObsRestricoes = getFieldFromLowestRip('obs_restricoes') || getFieldFromLowestRip('observacoes_restricoes');
            
            const checksRestricoes = document.querySelectorAll('input[name="restricoes[]"]');
            const inputObsRestricoes = document.getElementById('obs_restricoes');
            
            if (checksRestricoes.length > 0) {
                if (mockRestricoes === null) {
                    const formGrp = checksRestricoes[0].closest('.form-group');
                    const labelRestr = formGrp ? formGrp.querySelector('label') : null;
                    if (labelRestr) {
                        labelRestr.style.backgroundColor = '#fff3cd';
                        labelRestr.style.padding = '4px 8px';
                        labelRestr.style.borderRadius = '4px';
                        labelRestr.style.display = 'inline-block';
                        if (!labelRestr.querySelector('.hint-icon')) {
                            labelRestr.innerHTML += ' <span class="hint-icon" title="Dado ausente na base oficial. Se tiver o dado e desejar atualizá-lo, use o campo de Observações." style="cursor:help;">ℹ️</span>';
                        }
                    }
                    
                    checksRestricoes.forEach(c => c.disabled = true);
                    setHint(inputObsRestricoes, ''); 
                } else {
                    checksRestricoes.forEach(c => {
                        c.disabled = true;
                        if (mockRestricoes.map(x => x.toLowerCase()).includes(c.value.toLowerCase())) {
                            c.checked = true;
                        }
                    });

                    if (window.toggleRestricoes) window.toggleRestricoes();

                    const blocoObs = document.getElementById('bloco-obs-restricoes');
                    if (blocoObs && blocoObs.style.display !== 'none') {
                        setHint(inputObsRestricoes, mockObsRestricoes);
                    }
                }
            }

            // Limpa o interval pois os dados já foram carregados`;

if (!js.includes('// 7. RESTRIÇÕES')) {
    js = js.replace(target, restricoesLogic);
}

fs.writeFileSync('foco-02.js', js);
console.log('Restricoes patched successfully.');
